### PR TITLE
feat(dlm): incremenntal table structure synchronization

### PR DESCRIPTION
**Pull request by guowl3**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
Synchronize the table structure based on the results from the table structure comparator. After this PR is merged, DLM tasks will support automatic table structure synchronization for both MYSQL and OB data sources.

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/job/DLMJobParameters.java [hunk 1]
```diff
@@ -49,6 +49,8 @@ public class DLMJobParameters {
 
     private boolean deleteAfterMigration;
 
+    private boolean isTableStructureSyncEnabled;
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/job/DLMJobParameters.java:52]: may remove the `is` prefix

File: server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/DLMTableStructureSynchronizer.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                sourceConnectionConfig, tableNames);
+        DBStructureComparisonConfig targetConfig = initDBStructureComparisonConfig(
+                targetConnectionConfig, tableNames);
+        DefaultDBStructureComparator comparator = new DefaultDBStructureComparator();
+        List<DBObjectComparisonResult> compare = comparator.compare(sourceConfig, targetConfig);
+        StructureComparisonTaskResultEntity res = compare.get(0).toEntity(1L,
+                DialectType.OB_MYSQL);
+        if (StringUtils.isNotEmpty(res.getChangeSqlScript())) {
+            log.info("Start to sync target table structure,sqls={}", res.getChangeSqlScript());
+            try (Connection conn = targetConfig.getDataSource().getConnection()) {
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/DLMTableStructureSynchronizer.java:66]: Should we close or expire this DataSource?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/DataArchiveTask.java [hunk 3]
```diff
@@ -76,8 +63,17 @@ protected boolean doStart(JobContext context) throws Exception {
                 log.info("Job is terminated,jobIdentity={}", context.getJobIdentity());
                 break;
             }
-            if (parameters.getJobType() == JobType.MIGRATE) {
-                syncTable(tableIndex, parameters);
+            if (parameters.getJobType() == JobType.MIGRATE && parameters.isTableStructureSyncEnabled()) {
+                try {
+                    DLMTableStructureSynchronizer.sync(
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/DataArchiveTask.java:68]: The `DLMTableStructureSynchronizer.sync()` will not throw any exception if failed to execute sync SQL script. You may need refine the code logic.

**Review by smallsheeeep — CHANGES_REQUESTED** [review]

**Review by smallsheeeep — APPROVED** [review]

**Review by krihy — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**